validate.go

```diff
@@ -0,0 +1,35 @@
+package ykoath
+
+import (
+	"crypto/hmac"
+	"crypto/rand"
+)
+
+// Validate the OATH with a provided key
+func (o *OATH) Validate(s *Select, key []byte) (bool, error) {
+	algo, err := s.Hash()
+	if err != nil {
+		return false, err
+	}
+
+	mac := hmac.New(algo, key)
```

Comment:
Hm. If you really want to introduce a `Hash` function, you could potentially use the `Sum` / `Sum256` / `Sum512` functions from standard and return a slices on the array returned by them.

Reply:
I'm not sure I follow.

Reply:
I meant you could use the "sum" forms such as https://golang.org/pkg/crypto/sha256/#Sum256 - but I guess your way is more generic. Let's ignore this ...